econml/tests/test_automated_ml.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import unittest
+import pytest
+from sklearn.linear_model import LinearRegression, Lasso, \
+    LogisticRegression
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder, FunctionTransformer, \
+    PolynomialFeatures
+from sklearn.model_selection import KFold
+from econml.dml import *
+from econml.metalearners import *
+from econml.automated_ml import *
+from econml.drlearner import DRLearner
+import numpy as np
+from econml.utilities import shape, hstack, vstack, reshape, \
+    cross_product
+from econml.inference import BootstrapInference
+from contextlib import ExitStack
+from sklearn.ensemble import RandomForestRegressor, \
+    GradientBoostingRegressor, GradientBoostingClassifier
+import itertools
+from econml.sklearn_extensions.linear_model import WeightedLasso
+from econml.automated_ml import addAutomatedML
+from econml.tests.test_statsmodels import _summarize
+import econml.tests.utilities  # bugfix for assertWarns
+import copy
+import logging
+from econml.data.dgps import ihdp_surface_B
+from azureml.train.automl.exceptions import ClientException
+from azureml.core.authentication import AzureCliAuthentication
+import os
+
+AutomatedTLearner = addAutomatedML(TLearner)
+AutomatedSLearner = addAutomatedML(SLearner)
+AutomatedXLearner = addAutomatedML(XLearner)
+AutomatedDomainAdaptationLearner = \
+    addAutomatedML(DomainAdaptationLearner)
+AutomatedDRLearner = addAutomatedML(DRLearner)
+AutomatedDMLCateEstimator = addAutomatedML(DMLCateEstimator)
+AutomatedLinearDMLCateEstimator = addAutomatedML(LinearDMLCateEstimator)
+AutomatedSparseLinearDMLCateEstimator = \
+    addAutomatedML(SparseLinearDMLCateEstimator)
+AutomatedKernelDMLCateEstimator = addAutomatedML(KernelDMLCateEstimator)
+AutomatedNonParamDMLCateEstimator = \
+    addAutomatedML(NonParamDMLCateEstimator)
+AutomatedForestDMLCateEstimator = addAutomatedML(ForestDMLCateEstimator)
+
+# all solutions to underdetermined (or exactly determined) Ax=b are given by A⁺b+(I-A⁺A)w for some arbitrary w
+# note that if Ax=b is overdetermined, this will raise an assertion error
+
+
+def rand_sol(A, b):
+    """Generate a random solution to the equation Ax=b."""
+    assert np.linalg.matrix_rank(A) <= len(b)
+    A_plus = np.linalg.pinv(A)
+    x = A_plus @ b
+    return x + (np.eye(x.shape[0]) - A_plus @ A) @ np.random.normal(size=x.shape)
+
+
+AUTOML_SETTINGS_REG = {
+    'experiment_timeout_minutes': 1,
+    'enable_early_stopping': True,
+    'iteration_timeout_minutes': 1,
+    'max_cores_per_iteration': 1,
+    'n_cross_validations': 2,
+    'preprocess': False,
+    'featurization': 'off',
+    'enable_stack_ensemble': False,
+    'enable_voting_ensemble': False,
+    'primary_metric': 'normalized_mean_absolute_error',
```

Comment:
Why MAE instead of MSE?  I believe that for correctness at least DML should be using MSE in the final stage.

Reply:
Unfortunately, the only "evaluation metrics" I am able to user are spearman_correlation, normalized_root_mean_squared_error, r2_score, normalized_mean_absolute_error. I spoke to @vasilismsr about where using a specific error function should effect results and he suggest it should not. Would be interesting to investigate if we could get better performance that way however.

Reply:
I think minimizing normalized_root_mean_squared_error should be equivalent to minimizing MSE (it's the square root, but will have the same minimizer).  I thought that the weight trick in NonParamDML relied on MSE in a non-trivial way, but @vasilismsr would know for sure.

Reply:
I'll change it, I'll also rerun the notebooks with RMSE and see if it performs better.